teacher.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(id, age, specialization, name = 'Unknown')
+    super(id, age, name)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Kindly consider adding the parent_permission parameter for the teacher class as the absence of the parent_permission parameter in the initialize method will affect the behavior of your program later.

e.g

```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```


### Important
- [ ] (Optional) Consider removing the `id` as it is not required and necessary. This applies to the `student.rb` as well. They are both child classes of the `Person` class which will assign the `id`.

Reply:
Fixed